src/model.jl

```diff
@@ -180,11 +186,27 @@ function _make_sets(case::Case, storage::Union{Storage,Nothing})::Sets
                                * "Did you forget to linearize_gencost?")
     num_segments = convert(Int, maximum(case.gencost[:, NCOST])) - 1
     segment_idx = 1:num_segments
+    # Demand flexibility, additional info for input->bus conversion
+    demand_flexibility_enabled = isa(demand_flexibility, DemandFlexibility) && demand_flexibility.enabled
+    if demand_flexibility_enabled && demand_flexibility.cost_input_type == "BUS"
+        flexible_bus_str_up = names(demand_flexibility.flex_amt_up)[2:end]
+        flexible_bus_str_dn = names(demand_flexibility.flex_amt_dn)[2:end]
+        flexible_bus_id_up = [parse(Int64, bus) for bus in flexible_bus_str_up]
+        flexible_bus_id_dn = [parse(Int64, bus) for bus in flexible_bus_str_dn]
+        flexible_bus_idx_up = [bus_id2idx[bus] for bus in flexible_bus_id_up]
+        flexible_bus_idx_dn = [bus_id2idx[bus] for bus in flexible_bus_id_dn]
+        num_flexible_bus_up = length(flexible_bus_idx_up)
+        num_flexible_bus_dn = length(flexible_bus_idx_dn)
```

Comment:
We could probably have a nested `if`-`else` statement within the initially check for `demand_flexibility.enabled` that sets the demand flexibility sets for both `BUS` and `AREA`. For `BUS` the implementation would be similar to what you have now; for `AREA`, the demand flexibility sets could be set equal to those for the load buses. I believe this would remove the need for many of the `if`-`else` statements throughout `model.jl` (at least once the issue with `_make_sets` is resolved).

Reply:
Should we prevent user from specifying `AREA` and `BUS` with overlaps, i.e. some of the buses are within the `AREA` the demand flexibility of which is supposed to be defined in terms of `AREA`?

Reply:
I'm not sure I understand your question. The user only specifies `AREA` and `BUS` once at input (via `demand_flexibility_parameters.csv`). This helps determine the granularity of the demand flexibility data (e.g., if the user specifies `BUS`, all demand flexibility data that is input will be bus-level). Is that what you were asking?

Reply:
I see. So we don't allow the user to have both area and bus simultaneously, right? Hence, in certain situation, assuming in a use case, we would like to have all buses in CA plus some buses in WA with demand flexibility defined, the user has to choose `BUS` then put down all buses in CA and the subset of buses in WA in columns.

Reply:
Correct. I think the functionality you describe sounds great, but would that be better left for a future PR?

Reply:
@lanesmith Does it imply a major refactor of the code in this PR? If so, probably we should do it here.

Reply:
I don't think it would be a major refactor, it would just be a bigger change in the way that the granularity is input and we would need a lot more checks in `read_demand_flexibility` so that we prevent the overlap problem that you described above.

Reply:
Cool, then let's save it to a separate PR.